### PR TITLE
Support deserialize from file and mmap for faiss

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement

### CODE REVIEW

File: src/index/flat/flat.cc [hunk 1]
```diff
@@ -273,7 +273,28 @@ class FlatIndexNode : public IndexNode {
...
     Status
     DeserializeFromFile(const std::string& filename, const LoadConfig& config) override {
-        return Status::not_implemented;
+        std::string name = "";
+        if constexpr (std::is_same<T, faiss::IndexFlat>::value) {
+            name = "FLAT";
+        }
+        if constexpr (std::is_same<T, faiss::IndexBinaryFlat>::value) {
+            name = "BIN_FLAT";
+        }
```
Comment (liliu-z) [inline src/index/flat/flat.cc:282]: Where do we use this name?

Comment (yah01) [reply src/index/flat/flat.cc]: None, removed it

**Review by liliu-z — COMMENTED** [review]
/lgtm
/approve

Thanks

**Pull request merged**